File: 87.py
'''
@description: 密码强度等级
@param {type} 
@return: 
'''
while True:
    try:
        s=input()
        res=0
        if len(s)<=4:
            res+=5
        elif 5<=len(s)<=7:
            res+=10
        else:
            res+=25
        alpha_list=[]
        digit_list = []
        symbol_list=[]
        for i in s:
            if i.isalpha():
                alpha_list.append(i)
            if i.isdigit():
                digit_list.append(i)
            else:
                symbol_list.append(i)
        l=[0,0]
        for i in alpha_list:
            if ord(i) in range(97,123):
                l[0]=1
            if ord(i) in range(65,91):
                l[1]=1
        sum=l[0]+l[1]
        if sum==1:
            res+=10
        if sum==2:
            res+=20
        if len(digit_list)==1:
            res+=10
        if len(digit_list)>1:
            res+=20
        if len(symbol_list)==1:
            res+=10
        if len(symbol_list)>1:
            res+=25
        if len(alpha_list)!=0 and len(digit_list)!=0:
            res+=2
        elif len(alpha_list)!=0 and len(digit_list)!=0 and len(symbol_list)!=0:
            res+=3
        elif sum==2 and len(digit_list)!=0 and len(symbol_list)!=0:
            res+=5
        else:
            pass
        if res>=90:
            print('VERY_SECURE')
        elif res>=80:
            print('SECURE')
        elif res>=70:
            print('VERY_STRONG')
        elif res>=60:
            print('STRONG')
        elif res>=50:
            print('AVERAGE')
        elif res>=25:
            print('WEAK')
        else:
            print('VERY_WEAK')
    except:
        break

File: 87.c

/**
 * @description: 密码强度等级
 * @param {type} 
 * @return: 
 */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

char testbuf[100];
int testlen;
int daxieflag, xiaoxieflag;
int shuzicount;
int fuhaocount;
int fenshu;

int main(void)
{
    while(scanf("%s", testbuf) != EOF)
    {
        testlen = strlen(testbuf);
        daxieflag = 0;
        xiaoxieflag = 0;
        shuzicount = 0;
        fuhaocount = 0;
        fenshu = 0;

        if(testlen <= 4)
        {
            fenshu += 5;
        }
        else if(testlen <= 7)
        {
            fenshu += 10;
        }
        else
        {
            fenshu += 25;
        }

        for(int i = 0; i < testlen; i++)
        {
            if((testbuf[i] >= 'a')&&(testbuf[i] <= 'z'))
            {
                xiaoxieflag = 1;
            }
            else if((testbuf[i] >= 'A')&&(testbuf[i] <= 'Z'))
            {
                daxieflag = 1;
            }
            else if((testbuf[i] >= '0')&&(testbuf[i] <= '9'))
            {
                shuzicount++;
            }
            else if(((testbuf[i] >= 0x21)&&(testbuf[i] <= 0x2F))||((testbuf[i] >= 0x3A)&&(testbuf[i] <= 0x40))||\
            ((testbuf[i] >= 0x5B)&&(testbuf[i] <= 0x60))||((testbuf[i] >= 0x7B)&&(testbuf[i] <= 0x7E)))
            {
                fuhaocount++;
            }
        }

        if(daxieflag && xiaoxieflag)
        {
            fenshu += 20;
        }
        else if(daxieflag || xiaoxieflag)
        {
            fenshu += 10;
        }
        
        if(shuzicount > 1)
        {
            fenshu += 20;
        }
        else if(shuzicount == 1)
        {
            fenshu += 10;
        }

        if(fuhaocount > 1)
        {
            fenshu += 25;
        }
        else if(fuhaocount == 1)
        {
            fenshu += 10;
        }

        if(daxieflag && xiaoxieflag && shuzicount && fuhaocount)
        {
            fenshu += 5;
        }
        else if((daxieflag || xiaoxieflag) && shuzicount && fuhaocount)
        {
            fenshu += 3;
        }
        else if((daxieflag || xiaoxieflag) && shuzicount)
        {
            fenshu += 2;
        }

        if(fenshu >= 90)
        {
            printf("VERY_SECURE\n");
        }
        else if(fenshu >= 80)
        {
            printf("SECURE\n");
        }
        else if(fenshu >= 70)
        {
            printf("VERY_STRONG\n");
        }
        else if(fenshu >= 60)
        {
            printf("STRONG\n");
        }
        else if(fenshu >= 50)
        {
            printf("AVERAGE\n");
        }
        else if(fenshu >= 25)
        {
            printf("WEAK\n");
        }
        else
        {
            printf("VERY_WEAK\n");
        }
    }

    return 0;
}
